File: app/src/main/java/dev/ragnarok/fenrir/view/pager/WeakPicassoLoadCallback.java
package dev.ragnarok.fenrir.view.pager;

import androidx.annotation.NonNull;

import com.squareup.picasso3.Callback;

import java.lang.ref.WeakReference;

public class WeakPicassoLoadCallback implements Callback {

    private final WeakReference<Callback> mReference;

    public WeakPicassoLoadCallback(Callback baseCallback) {
        mReference = new WeakReference<>(baseCallback);
    }

    @Override
    public void onSuccess() {
        Callback callback = mReference.get();
        if (callback != null) {
            callback.onSuccess();
        }
    }

    @Override
    public void onError(@NonNull Throwable t) {
        Callback callback = mReference.get();
        if (callback != null) {
            callback.onError(t);
        }
    }

}

File: app_filegallery/src/main/kotlin/dev/ragnarok/filegallery/view/pager/WeakPicassoLoadCallback.kt
package dev.ragnarok.filegallery.view.pager

import com.squareup.picasso3.Callback
import java.lang.ref.WeakReference

class WeakPicassoLoadCallback(baseCallback: Callback) : Callback {
    private val mReference: WeakReference<Callback> = WeakReference(baseCallback)
    override fun onSuccess() {
        mReference.get()?.onSuccess()
    }

    override fun onError(t: Throwable) {
        mReference.get()?.onError(t)
    }
}